n2v/internals/N2V_DataWrapper.py

```diff
@@ -81,9 +85,39 @@ def __getitem__(self, i):
                 self.Y_Batches[indexing] = y_val
                 self.Y_Batches[indexing_mask] = 1
                 self.X_Batches[indexing] = x_val
+                
+                if self.structN2Vmask is not None:
+                    self.apply_structN2Vmask(self.X_Batches[j, ..., c], coords, self.dims, self.structN2Vmask)
 
         return self.X_Batches[idx], self.Y_Batches[idx]
 
+
+    def apply_structN2Vmask(self, patch, coords, dims, mask):
+        """
+        each point in coords corresponds to the center of the mask.
+        then for point in the mask with value=1 we assign a random value
+        """
+        coords = np.array(coords).astype(np.int)
+        ndim = mask.ndim
+        center = np.array(mask.shape)//2
+        ## leave the center value alone
+        mask[tuple(center.T)] = 0
+        ## displacements from center
+        dx = np.indices(mask.shape)[:,mask==1] - center[:,None]
+        ## combine all coords (ndim, npts,) with all displacements (ncoords,ndim,)
+        mix = (dx.T[...,None] + coords[None])
+        mix = mix.transpose([1,0,2]).reshape([ndim,-1]).T
+        ## stay within patch boundary
+        mix = mix.clip(min=np.zeros(ndim),max=np.array(patch.shape)-1).astype(np.uint)
+        ## replace neighbouring pixels with random values from flat dist
+        patch[tuple(mix.T)] = np.random.rand(mix.shape[0])*4 - 2
+        # patch[tuple(coords)] = 2
+
+
```

Comment:
Please remove empty code lines and commented code which is not used.

Reply:
Please :pray: ? :)

Reply:
ok fixed the remaining whitespace :)